core/src/test/java/org/elasticsearch/common/geo/GeoDistanceTests.java

```diff
@@ -88,15 +93,28 @@ public void testReadFromSerializationBWC() throws Exception {
         }
     }
 
-    public void testWriteToSerializationBWC() throws Exception {
+    public void testWriteToSerializationBWC53() throws Exception {
         GeoDistance geoDistance = randomFrom(GeoDistance.PLANE, GeoDistance.ARC);
         try (BytesStreamOutput out = new BytesStreamOutput()) {
-            out.setVersion(VersionUtils.randomVersionBetween(random(), Version.V_2_0_0, Version.V_5_3_2));
+            out.setVersion(VersionUtils.randomVersionBetween(random(), Version.V_5_3_0, Version.V_5_3_2));
             geoDistance.writeTo(out);
             try (StreamInput in = out.bytes().streamInput()) {
                 in.setVersion(out.getVersion());
-                GeoDistance copy = GeoDistance.readFromStream(in);
-                assertThat(copy, isOneOf(GeoDistance.PLANE, GeoDistance.ARC));
+                int ord = in.readVInt();
+                assertThat(ord, isIn(Arrays.asList(0, 1)));
```

Comment:
Yeah, I think the problem with the test here is that we don't make sure that nothing is left in the stream after we read it. That's why we didn't catch it here.